src/tile.cpp

```diff
@@ -396,94 +394,96 @@ void Tile::onUpdateTileItem(Item* oldItem, const ItemType& oldType, Item* newIte
 			}
 		}
 	} else if (oldItem->hasProperty(CONST_PROP_MOVEABLE) || oldItem->getContainer()) {
-		auto it = g_game.browseFields.find(this);
+		auto it = g_game.browseFields.find(getTile());
 		if (it != g_game.browseFields.end()) {
-			Cylinder* oldParent = oldItem->getParent();
+			auto oldParent = oldItem->getParent();
 			it->second->removeThing(oldItem, oldItem->getItemCount());
 			oldItem->setParent(oldParent);
 		}
 	}
 
-	const Position& cylinderMapPos = getPosition();
+	const Position& thingMapPos = getPosition();
 
 	SpectatorVec spectators;
-	g_game.map.getSpectators(spectators, cylinderMapPos, true);
+	g_game.map.getSpectators(spectators, thingMapPos, true);
 
 	// send to client
-	for (Creature* spectator : spectators) {
-		if (Player* spectatorPlayer = spectator->getPlayer()) {
-			spectatorPlayer->sendUpdateTileItem(this, cylinderMapPos, newItem);
+	for (const auto& spectator : spectators) {
+		if (auto spectatorPlayer = spectator->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
		if (const auto spectatorPlayer = spectator->getPlayer()) {
```